posts/tips-and-tricks-for-manually-scaling-a-global-dynamodb-table-from-an-aws-lambda.md

```diff
@@ -0,0 +1,128 @@
+---
+authors:
+- Dennis Sharpe
+tags:
+- AWS
+- Lambda
+- DynamoDB
+- Java
+date: 2020-11-16T15:52:00.000Z
+title: "Tips and Tricks for Manually Scaling a Global DynamoDB Table from an AWS Lambda"
+image:
+---
+# Objective
+Write an AWS Lambda that manually scales a global DynamoDB table
+
+# Why?
+DynamoDB tables can be set to automatically scale based on load. However, this process can take quite a while. We needed to scale up at a precise time to be ready for large batch loads.
+
+# MVP
+Scale a single (non-global) table
+- Turn off auto-scaling for the table (Provisioned)
+- Use the DynamoDB SDK v1
+- Explicitly call the SDK to set the RCU/WCU on the table
+- Initialize
+```java
+private ClientConfiguration clientConfiguration() {
+    final ClientConfiguration clientConfiguration = new ClientConfiguration();
+    clientConfiguration.setProxyHost(AWS_PROXY_HOST);
+    clientConfiguration.setProxyPort(Integer.valueOf(AWS_PROXY_PORT));
+    clientConfiguration.setClientExecutionTimeout(CLIENT_TIMEOUT);
+    clientConfiguration.setProtocol(Protocol.HTTP);
```

Comment:
May want to point out that these are environment variables that get injected unless you expect readers to already be familiar with this. Also, what values can they take on? Why would you set them to a certain value? Or dump links about this too.

Reply:
TBH i'd remove the proxy bits altogether; or specify that it may only be necessary when accessing dynamo from behind a proxy